invenio_rdm_records/services/communities/service.py

```diff
@@ -24,6 +25,38 @@ def create(self, identity, data, record, uow=None):
         pass
 
     @unit_of_work()
-    def delete(self, identity, id_, revision_id=None, uow=None):
+    def delete(self, identity, record_id, data, revision_id=None, uow=None):
         """Remove communities from the record."""
-        pass
+        record = self.record_cls.pid.resolve(record_id)
+        self.require_permission(identity, "remove_community_from_record", record=record)
+
+        valid_data, errors = self.schema.load(
+            data,
+            context={"identity": identity},
+            raise_errors=True,
+        )
+
+        removed_communities = []
+        errored_communities = []
+        community_ids = valid_data["communities"]
+        for community_id in community_ids:
+            if community_id not in record.parent.communities.ids:
```

Comment:
Cosmetic :) This is completely IMO.
To make the code more readable, what I have done on my WIP is to take advantage of exception and avoid `if/continue/break` (pseudo-code):
```
errors = {}
for community_id in community_ids:
    try:
        self._remove(...)
        # success actions
        ...
    except NotIncuded:
        errors[community_id] = msg
    except ...

    # record commit

    return errors
```

```
def _remove(...):
    not_included = community_id not in record.parent.communities.ids
    if not_included:
        raise NotIncuded
   
    ...
```

I would then transform the payload in the resource to make it:
```
errors: [ ... ]
```